File: Base/client/gui/GuiButtonCheck.java
package com.countrygamer.core.Base.client.gui;

import org.lwjgl.opengl.GL11;

import net.minecraft.client.Minecraft;
import net.minecraft.client.gui.GuiButton;
import net.minecraft.util.ResourceLocation;

import com.countrygamer.core.lib.CoreReference;

import cpw.mods.fml.relauncher.Side;
import cpw.mods.fml.relauncher.SideOnly;

@SideOnly(Side.CLIENT)
public class GuiButtonCheck extends GuiButton {

	private boolean isCheckmark = true;

	public GuiButtonCheck(int id, int x, int y, boolean isCheckmark) {
		super(id, x, y, 20, 20, "");
		this.isCheckmark = isCheckmark;
	}

	/**
	 * Draws this button to the screen.
	 */
	public void drawButton(Minecraft par1Minecraft, int x, int y) {
		if (this.visible) {
			par1Minecraft.getTextureManager().bindTexture(
					new ResourceLocation(CoreReference.MOD_ID,
							"textures/gui/buttons.png"));

			GL11.glColor4f(1.0F, 1.0F, 1.0F, 1.0F);
			boolean flag = x >= this.xPosition && y >= this.yPosition
					&& x < this.xPosition + this.width
					&& y < this.yPosition + this.height;
			int k = this.getHoverState(flag);

			this.drawTexturedModalRect(this.xPosition, this.yPosition,
					20 + (this.isCheckmark ? 0 : 20), k*20, this.width,
					this.height);
		}
	}

}

File: src/main/scala/com/countrygamer/cgo/client/gui/GuiButtonCheck.scala
package com.countrygamer.cgo.client.gui

import com.countrygamer.cgo.common.Origin
import com.countrygamer.cgo.common.lib.util.UtilRender
import cpw.mods.fml.relauncher.{Side, SideOnly}
import net.minecraft.client.Minecraft
import net.minecraft.client.gui.GuiButton
import net.minecraft.util.ResourceLocation
import org.lwjgl.opengl.GL11

/**
 *
 *
 * @author CountryGamer
 */
@SideOnly(Side.CLIENT)
class GuiButtonCheck(id: Int, x: Int, y: Int, val isCheckmark: Boolean)
		extends GuiButton(id, x, y, 20, 20, "") {

	val texture: ResourceLocation = new
					ResourceLocation(Origin.pluginID, "textures/gui/buttons.png")
	// Default Constructor

	// End Constructor

	// Other Constructors

	// End Constructors
	override def drawButton(minecraft: Minecraft, mouseX: Int, mouseY: Int): Unit = {
		if (this.visible) {
			UtilRender.bindResource(this.texture)
			GL11.glColor3f(1.0F, 1.0F, 1.0F)
			val isHoveredOn: Boolean = mouseX >= this.xPosition && mouseY >= this.yPosition &&
					mouseX < this.xPosition + this.width && mouseY < this.yPosition + this.height
			val k: Int = this.getHoverState(isHoveredOn)

			var u: Int = 0
			var v: Int = 0

			if (this.isCheckmark) {
				u += 20
			}

			this.drawTexturedModalRect(this.xPosition, this.yPosition, 20 + u, k * 20, this.width,
				this.height)

		}
	}
}